### PR TITLE
DAC6-2930 , DAC6-2928 : Added problem pages and navigation

### PR DESCRIPTION
**Pull request by AashuChauhanHmrc**

### CODE REVIEW

File: app/controllers/DifferentBusinessController.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                                            ) extends FrontendBaseController
+  with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = standardActionSets.identifiedUserWithData() {
+    implicit request =>
+      request.userAnswers.get(RegistrationInfoPage) match {
+        case Some(registrationInfo: RegistrationInfo) =>
+          val (name, address) = (Some(registrationInfo.name), Some(registrationInfo.address.asList))
+          Ok(view(appConfig.loginUrl, name, address))
+        case _ => Ok(view(appConfig.loginUrl, None, None))
```
Comment (C-Thomp) [inline app/controllers/DifferentBusinessController.scala:46]: minor - we could default the options in view constructor to remove the match

File: app/models/matching/RegistrationInfo.scala [hunk 2]
```diff
@@ -27,6 +27,9 @@ object RegistrationInfo {
 
   implicit val format: OFormat[RegistrationInfo] = Json.format[RegistrationInfo]
 
+
```
Comment (C-Thomp) [inline app/models/matching/RegistrationInfo.scala:30]: empty?

File: app/utils/CheckYourAnswersHelper.scala [hunk 3]
```diff
@@ -21,16 +21,23 @@ import pages.{BusinessHaveDifferentNamePage, DoYouHaveSecondContactPage, DoYouHa
...
 class CheckYourAnswersHelper(userAnswers: UserAnswers,
                              countryListFactory: CountryListFactory)(implicit val messages: Messages) {
 
-  def businessSection: Seq[SummaryListRow] = (userAnswers.get(IsRegisteredAddressInUkPage), userAnswers.get(DoYouHaveUTRPage)) match {
-    case (Some(true), _) => businessWithIDSection
-    case (Some(false), Some(true)) => businessWithIDSection
-    case (Some(false), _) => businessWithoutIDSection
-    case _ => Seq.empty[SummaryListRow]
+  def businessSection: Seq[SummaryListRow] = {
+    Console.println("Address :: 86768 :: "+ userAnswers)
```
Comment (C-Thomp) [inline app/utils/CheckYourAnswersHelper.scala:30]: remove printLl?

File: app/viewmodels/checkAnswers/SecondContactEmailSummary.scala [hunk 4]
```diff
@@ -26,6 +26,7 @@ import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
 import viewmodels.govuk.summarylist._
 import viewmodels.implicits._
 
+
```
Comment (C-Thomp) [inline app/viewmodels/checkAnswers/SecondContactEmailSummary.scala:29]: whitespace?

File: conf/app.routes [hunk 5]
```diff
@@ -23,7 +23,8 @@ GET         /account/sign-out                               controllers.auth.Aut
 GET         /register/problem/signed-out                    controllers.auth.SignedOutController.onPageLoad()
 
 GET         /unauthorised                                   controllers.UnauthorisedController.onPageLoad
-
+GET        /register/problem/different-business             controllers.DifferentBusinessController.onPageLoad()
```
Comment (C-Thomp) [inline conf/app.routes:26]: are these url's supposed to have /problem/ in the title?
often the full prototype url isn't needed

File: test/controllers/UnableToChangeBusinessControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,95 @@
+/*
```
Comment (C-Thomp) [inline test/controllers/UnableToChangeBusinessControllerSpec.scala:1]: double copyright??? 😆

File: test/controllers/UnableToChangeBusinessControllerSpec.scala [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+        val result = route(application, request).value
+
+        val view = application.injector.instanceOf[UnableToChangeBusinessView]
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(loginURL, None, None)(request, messages(application)).toString
+//=======
+//        contentAsString(result) mustEqual view()(request, messages(application)).toString
+//>>>>>>> 76969e2 (Added problem pages)
```
Comment (C-Thomp) [inline test/controllers/UnableToChangeBusinessControllerSpec.scala:91]: some merge diff here 👀

**Review by C-Thomp — CHANGES_REQUESTED** [review]
I've left a few comments that might need addressing, happy to work through them with you if you like but nothing major :)

File: conf/app.routes (hunk 5 above)
Comment (AashuChauhanHmrc) [reply conf/app.routes:26]: Yes as we are following same for other services as well like in mar and crs , so basically we are following a standard so hoplly keeping the url as per prototype would be a right approach.

**Review by C-Thomp — APPROVED** [review]
👍 thanks for those changes

**Pull request merged**